tests/model/test_ocrd_page.py

```diff
@@ -49,297 +53,415 @@
+
+    og = pcgts.get_Page().get_ReadingOrder().get_OrderedGroup()
+    children = og.get_AllIndexed()
+
+    # assert
+    assert isinstance(children[1], OrderedGroupIndexedType)
+    assert isinstance(children[21], UnorderedGroupIndexedType)
+    # empty all the elements in the first orederdGroupIndexed
+    children[1].set_RegionRefIndexed([])
+    # serialize apnd parse to see empty group converted
+    pcgts = parseString(to_xml(pcgts).encode('utf8'), silence=True)
+    og = pcgts.get_Page().get_ReadingOrder().get_OrderedGroup()
+    children = og.get_AllIndexed()
+    assert isinstance(children[1], RegionRefIndexedType)
+    assert isinstance(children[21], RegionRefIndexedType)
+
+
+def test_all_regions_without_reading_order():
+    """
+    https://github.com/OCR-D/core/pull/479
+    https://github.com/OCR-D/core/issues/240#issuecomment-493135797
+    """
+    with open(assets.path_to('gutachten/data/TEMP1/PAGE_TEMP1.xml'), 'r') as f:
+        pcgts = parseString(f.read().encode('utf8'), silence=True)
+
+    # act
+    pg = pcgts.get_Page()
+
+    # assert
+    assert len(pg.get_AllRegions()) == 65
+    assert len(pg.get_AllRegions(depth=0)) == 65
+    assert len(pg.get_AllRegions(depth=1)) == 45
+    assert len(pg.get_AllRegions(depth=2)) == 65
+    assert len(pg.get_AllRegions(depth=3)) == 65
+    assert len(pg.get_AllRegions(classes=['Separator'])) == 25
+    assert len(pg.get_AllRegions(classes=['Table'])) == 3
+    assert len(pg.get_AllRegions(classes=['Text'])) == 37
+    assert len(pg.get_AllRegions(classes=['Text'], depth=1)) == 17
+    assert len(pg.get_AllRegions(classes=['Text'], depth=2)) == 37
+
+
+def test_get_all_regions_invalid_order_raises_exeption():
+    # arrange
+    with open(assets.path_to('gutachten/data/TEMP1/PAGE_TEMP1.xml'), 'r') as f:
+        pg = parseString(f.read().encode('utf8'), silence=True).get_Page()
+
+    # act
+    with pytest.raises(Exception) as exc:
+        pg.get_AllRegions(order='random')
+
+    # assert
+    assert "Argument 'order' must be either 'document', 'reading-order' or 'reading-order-only', not 'random'" in str(exc.value)
+
+
+def test_get_all_regions_invalid_depth_raises_exeption():
+    # arrange
+    with open(assets.path_to('gutachten/data/TEMP1/PAGE_TEMP1.xml'), 'r') as f:
+        pg = parseString(f.read().encode('utf8'), silence=True).get_Page()
+
+    # act
+    with pytest.raises(Exception) as exc:
+        pg.get_AllRegions(depth=-1)
+
+    # assert
+    assert "Argument 'depth' must be an integer greater-or-equal 0, not '-1'" in str(exc.value)
+
+
+def test_all_regions_with_reading_order():
+    """
+    https://github.com/OCR-D/core/pull/479
+    https://github.com/OCR-D/core/issues/240#issuecomment-493135797
+    """
+
+    # arrange
+    with open(assets.path_to('gutachten/data/TEMP1/PAGE_TEMP1.xml'), 'r') as f:
+        pg = parseString(f.read().encode('utf8'), silence=True).get_Page()
+
+    # assert
+    assert len(pg.get_AllRegions(order='reading-order-only')), 40
```

Comment:
```suggestion
    assert len(pg.get_AllRegions(order='reading-order-only')) == 40
```

Or is this another pytest-specific syntax for `assert`?

Reply:
corrected in 55fd25a5